assembly/NFT/NFT.ts

```diff
@@ -0,0 +1,226 @@
+ * @return {string}
+ */
+export function symbol(_: string): string {
+  return _symbol;
+}
+
+/**
+ * Return the token URI (external link written in NFT where pictures or others a stored)
+ * @param {string} tokenId
+ * @return {string}
+ */
+export function tokenURI(tokenId: string): string {
+  if (Storage.has(baseURIKey)) {
+    return Storage.get(baseURIKey) + tokenId;
+  } else {
+    generateEvent(`NFT not setted yet`);
+    return '';
+  }
+}
+
+/**
+ * Return base URI
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+export function baseURI(_: string): string {
+  if (Storage.has(baseURIKey)) {
+    return Storage.get(baseURIKey);
+  } else {
+    generateEvent(`NFT not setted yet`);
+    return '';
+  }
+}
+
+/**
+ * Return the max supply possible
+ *@param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+export function limitSupply(_: string): string {
+  return _maxSupply;
+}
+
+/**
+ * Return the current counter, if 10 NFT minted, returns '10'.
+ *  @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ *  @return {string}
+ */
+export function currentSupply(_: string): string {
+  if (Storage.has(counterKey)) {
+    return Storage.get(counterKey);
+  } else {
+    generateEvent(`NFT not setted yet`);
+    return '';
+  }
+}
+
+/**
+ *
+ * Return the tokenId's owner
+ * @param {string} tokenId
+ * @return {string}
+ */
+export function ownerOf(tokenId: string): string {
+  if (Storage.has(ownerTokenKey + tokenId.toString())) {
+    return Storage.get(ownerTokenKey + tokenId.toString());
+  } else {
+    generateEvent(`NFT not minted yet`);
+    return '';
+  }
+}
+
+// ==================================================== //
+// ====                 TRANSFER                   ==== //
+// ==================================================== //
+/**
+ *
+ * The to address becomes the owner of the next token (if current tokenID = 10, will mint 11 )
+ * Check if max supply is not reached
+ * @param {string} args - byte string containing an owner's account (Address).
+ * @return {string}
+ */
+export function mint(args: string): string {
+  if (u32(parseInt(limitSupply(''))) > u32(parseInt(currentSupply('')))) {
+    const addr = Address.fromByteString(args);
+    _increment('');
+    const tokenID: string = currentSupply('');
+    const key = ownerTokenKey + tokenID;
+    Storage.set(key, addr.toByteString());
+    generateEvent(`tokenId ${tokenID} minted to ${addr.toByteString()} `);
+  } else {
+    generateEvent(`Max supply reached`);
+  }
+  return 'Minted';
+}
+
+/**
+ * Increment the NFT counter
+ * @param {string} _ - unused see https://github.com/massalabs/massa-sc-std/issues/18
+ * @return {string}
+ */
+function _increment(_: string): string {
+  const add1: u32 = 1;
+  const incr = u32(parseInt(Storage.get(counterKey))) + add1;
```

Comment:
```suggestion
  const incr = u32(parseInt(Storage.get(counterKey))) ++;
```
This'd be clearer I think

Reply:
I agree

Reply:
"The operand of an increment or decrement operator must be a variable or a property access i implement" this